src/coreclr/gc/gc.cpp

```diff
@@ -12141,6 +12141,13 @@ void region_free_list::add_region_descending (heap_segment* region, region_free_
     to_free_list[kind].add_region_in_descending_order (region);
 }
 
+bool region_free_list::is_on_free_list (heap_segment* region, region_free_list free_list[count_free_region_kinds])
```

Comment:
assume the free list is usually not long so the search is fast?

Reply:
There is no searching - if a region is on a free list, it contains a pointer to that free list. So the check is constant time.

Reply:
ah nice.